### PR TITLE
Justin Tracy Code Review 1

### PR DESCRIPTION
**Pull request by JustinTracy**

### CODE REVIEW

File: app/src/main/java/com/memrepo/MainActivity.kt [hunk 1]
```diff
@@ -8,9 +6,7 @@ import androidx.compose.foundation.layout.*
 import androidx.compose.foundation.shape.RoundedCornerShape
 import androidx.compose.material.*
 import androidx.compose.runtime.*
-import androidx.compose.ui.Alignment
-import androidx.compose.ui.ExperimentalComposeUiApi
-import androidx.compose.ui.Modifier
+import androidx.compose.ui.*
```
Comment (JustinTracy) [inline app/src/main/java/com/memrepo/MainActivity.kt:9]: These imports can be consolidated.

File: app/src/main/java/com/memrepo/MainActivity.kt [hunk 2]
```diff
@@ -22,25 +18,19 @@ class MainActivity : ComponentActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContent {
-            setContent {
```
Comment (JustinTracy) [inline app/src/main/java/com/memrepo/MainActivity.kt:25]: got rid of extra setContent

File: app/src/main/java/com/memrepo/dao/ICardDAO.kt [hunk 3]
```diff
@@ -16,5 +16,5 @@ interface ICardDAO {
     fun updateNoteCard(vararg notecards: NoteCard)
 
     @Delete
-    fun delete(notecards: NoteCard)
```
Comment (JustinTracy) [inline app/src/main/java/com/memrepo/dao/ICardDAO.kt:19]: Changing this function name to be more consistent with the other functions in the interface might be a good idea

File: app/src/main/java/com/memrepo/dto/NoteCard.kt [hunk 4]
```diff
@@ -5,17 +5,18 @@ import androidx.room.Entity
 import androidx.room.PrimaryKey
 
 @Entity
-data class NoteCard(@PrimaryKey(autoGenerate = true) val cardID : Int,@ColumnInfo("title") val title : String,@ColumnInfo("snippet") val snippet : String){
+data class NoteCard(
+    @PrimaryKey(autoGenerate = true) val cardID: Int,
+    @ColumnInfo(name = "title") val title: String,
+    @ColumnInfo(name = "snippet") val snippet: String
```
Comment (JustinTracy) [inline app/src/main/java/com/memrepo/dto/NoteCard.kt:11]: I think putting each column on a new line improves readability

File: app/src/main/java/com/memrepo/dto/NoteCard.kt [hunk 5]
```diff
@@ -5,17 +5,18 @@ import androidx.room.Entity
...
+    //var snippetDisplayList = snippetDisplay.split(" ").toList() create a list by splitting string on space
+    //var snippetDisplayList : List<String> = listOf<String>(this.snippetDisplay.split(" ").toString())
 
-        //var snippetDisplayList = snippetDisplay.split(" ").toList() create a list by splitting string on space
-        //var snippetDisplayList : List<String> = listOf<String>(this.snippetDisplay.split(" ").toString())
-
-        override fun toString(): String {
-                return snippetDisplay
-        }
+    override fun toString() = snippetDisplay
```
Comment (JustinTracy) [inline app/src/main/java/com/memrepo/dto/NoteCard.kt:21]: if you set toString() to a variable, it will automatically return it

**Comment by JustinTracy** [comment]
In addition to the comments I made above, I also fixed a few formatting things to make the code just a bit easier to read

**Comment by JustinTracy** [comment]
Program Analysis

- I think that the application is a really cool idea and that the team members have it in a good place development wise for the first sprint.  The code was for the most part really clear and easy to understand.  The file structure was good as well and I was always able to navigate to where I needed to go without issue.  I am excited to see how this application progresses in the future.
- This program was available on Github on time
- There were a bunch of helpful comments throughout the codebase that made understanding more complex logic easier
- The program does compile

Links to commits I made from my own group:
- https://github.com/OneList/OneList/commit/171ee5ef2167e33f3dbfe3fb39c5f01d317c29a4
- https://github.com/OneList/OneList/commit/65617f014325a359186215b1f0c1ce03fd8a401d
- https://github.com/OneList/OneList/commit/c6024a3f2a0907f55808e4d99d12e997e5b3e844

File: app/src/main/java/com/memrepo/MainActivity.kt (hunk 1 above)
Comment (discospiff) [reply app/src/main/java/com/memrepo/MainActivity.kt:9]: I'd leave this as it were. Wildcard imports (*) are considered bad practice for a couple of reasons:

-Less documentation on what classes are being imported.
-If you have two wildcard imports, and they have a class with the same name, there will be a collision. Example: java.util.* and java.sql.* both have a class named Date. Different classes, same name. So, if you're using wildcards, which Date class should be used?
- It makes no difference in runtime performance, as the compiler will automatically optimize imports.

File: app/src/main/java/com/memrepo/MainActivity.kt (hunk 2 above)
Comment (discospiff) [reply app/src/main/java/com/memrepo/MainActivity.kt:25]: Good.

File: app/src/main/java/com/memrepo/MainActivity.kt [hunk 6]
```diff
@@ -49,29 +39,13 @@ class MainActivity : ComponentActivity() {
...
-                    Modifier
-                        .fillMaxWidth()
-                        .fillMaxHeight()
-                ) {
-                    Column {
-                        // Contents of the Bottom Sheet come from this component
-                        AddSnippet(bottomSheetScaffoldState)
-                    }
+                Box(Modifier.fillMaxWidth().fillMaxHeight()) {
+                    Column { AddSnippet(bottomSheetScaffoldState) }
```
Comment (discospiff) [inline app/src/main/java/com/memrepo/MainActivity.kt:43]: Formatting can be performed automatically, and consistently, by the IDE.  No need to do it manually.  Formatting code is really programmer's preference anyway, and doesn't necessarily decrease technical debt.

File: app/src/main/java/com/memrepo/MainActivity.kt [hunk 7]
```diff
@@ -169,31 +141,29 @@ class MainActivity : ComponentActivity() {
...
+                    onClick = {
+                        // Logic to save snippet
+                        /* TODO */
+                        coroutineScope.launch { bottomSheetScaffoldState.bottomSheetState.collapse() }
+                        keyBoardController?.hide()
+                    },
+                    modifier = Modifier.align(Alignment.CenterHorizontally)
+                ) {
+                    Text("Save")
+                }
```
Comment (discospiff) [inline app/src/main/java/com/memrepo/MainActivity.kt:156]: What's happening with these changes?

File: app/src/main/java/com/memrepo/dao/ICardDAO.kt (hunk 3 above)
Comment (discospiff) [reply app/src/main/java/com/memrepo/dao/ICardDAO.kt:19]: Actually...   deleting NoteCards  is given by the implicit parameter name and type.  I'd leave this as it is.

File: app/src/main/java/com/memrepo/dto/NoteCard.kt (hunk 5 above)
Comment (discospiff) [reply app/src/main/java/com/memrepo/dto/NoteCard.kt:21]: Good.

**Comment by discospiff** [comment]
From this review, I'd remove the extra setContent.

Formatting and organizing imports can be handled automatically by the IDE, so I'd stick with that in lieu of manual changes.

**Pull request closed without merging**